packages/solana/lib/src/rpc/extension.dart

```diff
@@ -32,18 +32,44 @@ extension RpcClientExt on RpcClient {
       limit: limit,
       commitment: commitment,
     );
-    final transactions = await Future.wait(
-      signatures.map(
-        (s) => getTransaction(
-          s.signature,
-          commitment: commitment,
-          encoding: Encoding.jsonParsed,
-        ),
-      ),
+
+    return getMultipleTransactions(
+      signatures,
+      commitment: commitment,
+      encoding: Encoding.jsonParsed,
+    );
+  }
+
+  /// Get multiple transactions in 1 call.
+  ///
+  /// Gets one transaction for each signature in the [signatures] list.
+  ///
+  /// The [encoding] and [commitment] parameters are "passed as is"
+  /// to the internal call to [RpcClient.getTransaction()]
+  Future<List<TransactionDetails>> getMultipleTransactions(
+    List<TransactionSignatureInformation> signatures, {
+    Commitment? commitment,
+    Encoding? encoding,
+  }) async {
+    final response = await _jsonRpcClient.bulkRequest(
+      'getTransaction',
+      signatures
+          .where((s) => s.err == null)
```

Comment:
Why do you skip transactions with errors?

Reply:
I havent thought this through honestly. Good catch.